aptos-move/block-executor/src/counters.rs

```diff
@@ -22,12 +24,52 @@ pub static SPECULATIVE_ABORT_COUNT: Lazy<IntCounter> = Lazy::new(|| {
     .unwrap()
 });
 
-/// Count of times a transaction got suspended due to an estimated r/w dependency.
-pub static DEPENDENCY_SUSPEND_COUNT: Lazy<IntCounter> = Lazy::new(|| {
-    register_int_counter!(
-        "aptos_execution_dependency_suspend_count",
-        "Count write estimates encountered when reading in parallel execution \
-        (typically leading to a suspension / waiting)"
+pub static VM_INIT_SECONDS: Lazy<Histogram> = Lazy::new(|| {
+    register_histogram!(
+        // metric name
+        "aptos_execution_vm_init_seconds",
+        // metric description
+        "The time spent in seconds in initializing the VM in the block executor",
+        exponential_buckets(/*start=*/ 1e-3, /*factor=*/ 2.0, /*count=*/ 20).unwrap(),
+    )
+    .unwrap()
+});
+
+pub static TASK_VALIDATE_SECONDS: Lazy<Histogram> = Lazy::new(|| {
+    register_histogram!(
+        // metric name
+        "aptos_execution_task_validate_seconds",
+        // metric description
+        "The time spent in task validation in Block STM",
+        exponential_buckets(/*start=*/ 1e-3, /*factor=*/ 2.0, /*count=*/ 20).unwrap(),
+    )
+    .unwrap()
+});
+
+pub static TASK_EXECUTE_SECONDS: Lazy<Histogram> = Lazy::new(|| {
+    register_histogram!(
+        // metric name
+        "aptos_execution_task_execute_seconds",
+        // metric description
+        "The time spent in seconds for task execution in Block STM",
+        exponential_buckets(/*start=*/ 1e-3, /*factor=*/ 2.0, /*count=*/ 20).unwrap(),
+    )
+    .unwrap()
+});
+
+pub static DEPENDENCY_WAIT_SECOND: Lazy<Histogram> = Lazy::new(|| {
+    register_histogram!(
+        "aptos_execution_dependency_wait",
+        "The time spent in waiting for dependency in Block STM",
+        exponential_buckets(/*start=*/ 1e-3, /*factor=*/ 2.0, /*count=*/ 20).unwrap(),
+    )
+    .unwrap()
+});
+
+pub static GET_STATE_VALUE_SECONDS: Lazy<Histogram> = Lazy::new(|| {
```

Comment:
hmmm, will this affect performance?

this is probably a value under lock, so every state value fetch (even if not going to storage) will have a synchronization point?

@gelash  for thoughts?

Reply:
Yes, I thought about this - I had run some benchmark with and without these counters and didn't see any difference (which agrees with current Forge result as well). In fact, there are several counters on the storage side as well that measures API latencies in similar ways, so this is hardly adding any additional overhead.

Reply:
I also have concern here.
(having counter in storage side is mostly fine because storage is usually have higher latency than the overhead of the counter, but here it might not be the case)

BTW let's write a per-cpu counter/histogram?

Reply:
>BTW let's write a per-cpu counter/histogram?
@grao1991 - Can you clarify here? Do you mean per thread counter?

Reply:
I meant per cpu. (to save the lock)

in pseudocode:

int v[NUM_CPU];
void increment() {
  ++v[get_cpu_core_id()];
}

int collect() {
  return sum(v);
}

Reply:
I removed this for now - have added another counter on the storage side and that should be sufficient for us.